plugins/dmxcontrol3/exportTests/channel-numbers.js

```diff
@@ -0,0 +1,152 @@
+  const parser = new xml2js.Parser();
+  const parseString = promisify(parser.parseString);
+
+  const fixture = exportFile.fixtures[0];
+  const mode = fixture.modes.find(mode => mode.shortName === exportFile.mode);
+  const channelCount = mode.channels.length;
+
+  /** @type {!object.<number, Array.<Range>} */
+  const usedChannelRanges = {};
+
+  const errors = [];
+
+  const xml = await parseString(exportFile.content);
+  xml.device.functions.forEach(xmlFunction => findChannels(xmlFunction, -1));
+
+  checkUsedChannels();
+
+  if (errors.length > 0) {
+    throw errors;
+  }
+
+  /**
+   * Recursively searches the given XML tree for tags with dmxchannel attributes and capabilities.
+   * @param {!XMLElement} xmlNode A single XML node.
+   * @param {!number} currentChannelIndex The index of the channel if the xmlNode is inside a function associated to a channel. Else, it's -1.
+   */
+  function findChannels(xmlNode, currentChannelIndex) {
+    if (xmlNode.$) {
+      const indexAttributes = [`dmxchannel`, `finedmxchannel`, `ultradmxchannel`, `ultrafinedmxchannel`];
+      indexAttributes.filter(attr => attr in xmlNode.$).forEach(attr => {
+        const channelIndex = xmlNode.$[attr];
+
+        if (!(channelIndex in usedChannelRanges)) {
+          usedChannelRanges[channelIndex] = [];
+        }
+
+        if (attr === `dmxchannel`) {
+          currentChannelIndex = channelIndex;
+        }
+
+        if (parseInt(channelIndex) >= channelCount) {
+          errors.push(`${attr}=${channelIndex} is out of range (maximum: ${channelCount - 1}).`);
+        }
+      });
+
+      if (`mindmx` in xmlNode.$) {
+        // xmlNode is a capability
+
+        const mindmx = parseInt(xmlNode.$.mindmx);
+        const maxdmx = parseInt(xmlNode.$.maxdmx);
+        const range = new Range([Math.min(mindmx, maxdmx), Math.max(mindmx, maxdmx)]);
+
+        addCapability(range, currentChannelIndex);
+      }
+    }
+
+    for (const tagname of Object.keys(xmlNode)) {
+      if (tagname !== `$`) {
+        for (const child of xmlNode[tagname]) {
+          findChannels(child, currentChannelIndex);
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks the given capability range and adds it to the channel's ranges.
+   * @param {!Range} range A valid Range instance (start <= end).
+   * @param {!number} channelIndex The index of the channel that contains this capability.
+   */
+  function addCapability(range, channelIndex) {
+    if (channelIndex === -1) {
+      errors.push(`Capability ${range} is not inside a channel function.`);
+    }
+    else if (range.start < 0 || range.end > 255) {
+      errors.push(`Capability ${range} in channel #${channelIndex + 1} is out of the allowed 0…255 range.`);
+    }
+    else {
+      const existingRanges = usedChannelRanges[channelIndex];
+
+      if (range.overlapsWithOneOf(existingRanges)) {
+        errors.push(`Capability ${range} in channel #${channelIndex + 1} overlaps with other capabilities.`);
```

Comment:
```suggestion
        errors.push(`Capability ${range} in channel ${channelIndex + 1} overlaps with other capabilities.`);
```

(without `#` to prevent GitHub from linking it to an issue / PR)